### PR TITLE
Implement cannot-be-a-base-url flag

### LINKED ISSUE #459: `URL#toString()` is incorrect for URLs that cannot be used as base URLs
URL parsing and stringification do not handle the `cannot be a base URL` flag (as far as I understand).
This result in `{ scheme: "about", path: ["blank"] }` to be stringified as `about:/blank` instead of `about:blank`.

### PR DESCRIPTION
**Pull request by vemolista**
Resolves #459 

Spec:
https://url.spec.whatwg.org/#concept-base-url
https://url.spec.whatwg.org/#url-serializing

### CODE REVIEW

**Comment by kasperisager** [comment]
I dug through the spec again and sorted out the (hopefully) last quirks so I'll punt this back to you, @vemolista. If there's nothing else you'd like to add then feel to send this off for review.

File: packages/alfa-url/src/url.ts [hunk 1]
```diff
@@ -34,10 +35,25 @@ export class URL implements Equatable, Hashable, Serializable<URL.JSON> {
...
+    return URL.parse("https://example.com").get();
+  }
+
+  /**
+   * @see https://tools.ietf.org/html/rfc6694#section-3
+   */
+  public static blank(): URL {
+    return URL.parse("about:blank").get();
+  }
+
```
Comment (Jym77) [inline packages/alfa-url/src/url.ts:56]: These two seem to be unused 🤔

File: packages/alfa-url/src/url.ts [hunk 2]
```diff
@@ -354,11 +400,33 @@ export namespace URL {
           // "#" which we need to remove.
           Option.of(hash)
             .reject(isEmpty)
-            .map((hash) => hash.replace(/^#/, ""))
+            .map((hash) => hash.replace(/^#/, "")),
+
+          // The URL cannot be used as a base URL when the scheme either isn't
+          // special or the pathname doesn't start with a leading "/".
+          !(isSpecialScheme(scheme) || pathname[0] === "/")
```
Comment (Jym77) [inline packages/alfa-url/src/url.ts:407]: Comment and code don't match.
Code test "scheme not special **and** path doesn't start with /" but comment says **or**.
Given that tests are working, I guess that code is correct. Thus said, I also suggest to De Morgan-ise it to better match the comment.
```suggestion
          // The URL cannot be used as a base URL when the scheme isn't
          // special and the pathname doesn't start with a leading "/".
          !isSpecialScheme(scheme) && pathname[0] !== "/"
```

**Review by Jym77 — CHANGES_REQUESTED** [review]
Small editorial cleanup.

File: packages/alfa-url/src/url.ts (hunk 1 above)
Comment (kasperisager) [reply packages/alfa-url/src/url.ts:56]: `URL.blank()` currently is, but `URL.example()` isn't.

File: packages/alfa-url/src/url.ts (hunk 2 above)
Comment (kasperisager) [reply packages/alfa-url/src/url.ts:407]: Good catch!

**Review by Jym77 — DISMISSED** [review]

**Review by kasperisager — APPROVED** [review]

**Review by kasperisager — APPROVED** [review]

**Pull request merged**